src/runtime/program/vote/execute.zig

```diff
@@ -428,6 +429,61 @@ fn executeAuthorizeChecked(
     );
 }
 
+/// Agave https://github.com/anza-xyz/agave/blob/24e62248d7a91c090790e7b812e23321fa1f53b1/programs/vote/src/vote_processor.rs#L114-L118
+fn executeUpdateValidatorIdentity(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+) InstructionError!void {
+    try ic.info.checkNumberOfAccounts(2);
+
+    var new_identity = try ic.borrowInstructionAccount(
+        @intFromEnum(vote_instruction.UpdateVoteIdentity.AccountIndex.new_identity),
+    );
+    defer new_identity.release();
+
+    try updateValidatorIdentity(
+        allocator,
+        ic,
+        vote_account,
+        new_identity.pubkey,
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/24e62248d7a91c090790e7b812e23321fa1f53b1/programs/vote/src/vote_state/mod.rs#L722
+///
+/// Update the node_pubkey, requires signature of the authorized voter
+fn updateValidatorIdentity(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    new_identity: Pubkey,
+) InstructionError!void {
+    const versioned_state = try vote_account.deserializeFromAccountData(
+        allocator,
+        VoteStateVersions,
+    );
+
+    var vote_state = versioned_state.convertToCurrent(allocator) catch {
+        // TODO okay to convert out of memory to InvalidAccountData?
+        return InstructionError.InvalidAccountData;
+    };
+    defer vote_state.deinit();
+
+    // current authorized withdrawer must say "yay"
```

Comment:
can we change these comments to something more descriptive instead of "must say 'yay'" XD -- tbh maybe we dont even need the comments its pretty clear your checking the signer

Reply:
I agree. it was blind copy from Agave. Removed instead [bc11aa3](https://github.com/Syndica/sig/pull/611/commits/bc11aa3e37295871544527b8eefbacf7a04707da)

Reply:
Brought the comment back and used one `if` [f176961](https://github.com/Syndica/sig/pull/611/commits/f1769618697d46e38a8d4a5bad427cd1072c3edd)